tests/Neo.UnitTests/Ledger/UT_MemoryPool.cs

```diff
@@ -241,6 +242,172 @@ public async Task BlockPersistAndReverificationWillAbandonTxAsBalanceTransfered(
+            _unit.TryAdd(mp4, engine.Snapshot);
+
+            var tx4 = CreateTransactionWithFeeAndBalanceVerify(txFee);  // in-block tx4 and tx5 don't conflict with anyone
+            var tx5 = CreateTransactionWithFeeAndBalanceVerify(txFee);
+            var mp5 = CreateTransactionWithFeeAndBalanceVerify(txFee);  // mp5 conflicts with in-block tx4 and tx5 => mp5 should be removed from pool after persist
+            mp5.Attributes = new TransactionAttribute[] { new Conflicts() { Hash = tx4.Hash }, new Conflicts() { Hash = tx5.Hash } };
+            _unit.TryAdd(mp5, engine.Snapshot);
+
+            var mp6 = CreateTransactionWithFeeAndBalanceVerify(txFee);  // mp6 doesn't conflict with anyone and noone conflicts with mp6 => mp6 should be left in the pool after persist
+            _unit.TryAdd(mp6, engine.Snapshot);
+
+            _unit.SortedTxCount.Should().Be(6);
+            _unit.UnverifiedSortedTxCount.Should().Be(0);
+
+            // Act: persist block and reverify all mempooled txs.
+            var block = new Block
+            {
+                Header = new Header(),
+                Transactions = new Transaction[] { tx1, tx2, tx3, tx4, tx5 },
+            };
+            _unit.UpdatePoolForBlockPersisted(block, engine.Snapshot);
+
+            // Assert: conflicting txs should be removed from the pool; the only mp6 that doesn't conflict with anyone should be left.
+            _unit.SortedTxCount.Should().Be(1);
+            _unit.GetSortedVerifiedTransactions().Select(tx => tx.Hash).Should().Contain(mp6.Hash);
+            _unit.UnverifiedSortedTxCount.Should().Be(0);
+
+            // Cleanup: revert the balance.
+            await NativeContract.GAS.Burn(engine, UInt160.Zero, txFee * 6);
+            _ = NativeContract.GAS.Mint(engine, UInt160.Zero, balance, true);
+        }
+
+        [TestMethod]
+        public async Task TryAdd_AddRangeOfConflictingTransactions()
+        {
+            // Arrange: prepare mempooled txs that have conflicts.
+            long txFee = 1;
+            var snapshot = GetSnapshot();
+            BigInteger balance = NativeContract.GAS.BalanceOf(snapshot, senderAccount);
+            ApplicationEngine engine = ApplicationEngine.Create(TriggerType.Application, null, snapshot, settings: TestBlockchain.TheNeoSystem.Settings, gas: long.MaxValue);
+            engine.LoadScript(Array.Empty<byte>());
+            await NativeContract.GAS.Burn(engine, UInt160.Zero, balance);
+            _ = NativeContract.GAS.Mint(engine, UInt160.Zero, 100, true); // balance enough for all mempooled txs
+
+            var mp1 = CreateTransactionWithFeeAndBalanceVerify(txFee);  // mp1 doesn't conflict with anyone and not in the pool yet
+
+            var mp2 = CreateTransactionWithFeeAndBalanceVerify(txFee);  // mp2 conflicts with mp1 and has the same network fee
+            mp2.Attributes = new TransactionAttribute[] { new Conflicts() { Hash = mp1.Hash } };
+            _unit.TryAdd(mp2, engine.Snapshot);
+
```

Comment:
Please add here.
Rename `mp2` to `mp2-1`
```cs
var mp2-2 = CreateTransactionWithFeeAndBalanceVerify(txFee);  // mp2-2 also conflicts with mp1 and has the same network fee as mp1 and mp2-1
            mp2.Attributes = new TransactionAttribute[] { new Conflicts() { Hash = mp1.Hash } };
            _unit.TryAdd(mp2, engine.Snapshot);
```

Reply:
It is almost like this check here:
```cs

            // Act & Assert: try to add conlflicting transactions to the pool.
            _unit.TryAdd(mp1, engine.Snapshot).Should().Be(VerifyResult.HasConflicts); // mp1 conflicts with mp2 and mp3 but has lower network fee than mp3 => mp1 fails to be added
            _unit.SortedTxCount.Should().Be(2);
            _unit.GetVerifiedTransactions().Should().Contain(new List<Transaction>() { mp2, mp3 });
```

However, we ensure that the equal condition is also working.

Reply:
@vncoelho, I've added this testcase, it works as expected, please, see the new commit. Do you have a question about it?

Reply:
It works as expected. Thanks.